ftw/contacts/resources/contactfolder_listing.js

```diff
@@ -0,0 +1,81 @@
+$(function() {
+    ContactFolderListing.init();
+});
+var ContactFolderListing = (function($){
+    "use strict"
+    var self = {};
+    var $contactsContainer;
+    var $loadMoreButton;
+    var $alphabeticalContainer;
+    var $searchInput;
+    var index = 0;
+    var step = 20;
+    var letter = '';
+    var maxContacts = 0;
+    var searchableText = ''
+
+    var init = function(){
+        $contactsContainer = $('#contact-folder-view .contactFolderContactsListig');
+        $loadMoreButton = $('#contact-folder-view .contactFolderLoadMoreContacts');
+        $searchInput = $('#contact-folder-view input#contactFolderSearchGadget');
+
+        $loadMoreButton.on('click', function(e) {reloadView();});
+        $('.contactFolderAlphabeticalSearch div.letter').on('click', function(e) { letterClick($(this)); });
+        $searchInput.on('keyup', function() {
+            updateSearch($(this).val());
+        });
+
+        reloadView();
+    }
+    var reloadView = function() {
+        $.getJSON( "@@reload", {
+                index_from: index,
+                index_to: index + step,
+                letter: letter,
+                searchable_text: searchableText}, function( data ){
+
+            // contacts
+            $contactsContainer.append( $(data.contacts) );
+
+            index = index + step;
+            setMoreButtonVisible(index < data.max_contacts);
+
+            // letters
+            $('.contactFolderAlphabeticalSearch').html( data.letters );
+            $('.contactFolderAlphabeticalSearch div.letter').on('click', function(e) {
+                letterClick($(this));
+            });
+        });
+    }
+    var letterClick = function(button){
+        var $this = $(button);
+        if ( !$this.hasClass('withContent') && !$this.hasClass('current')){
+            return;
+        }
+        index = 0;
+        $contactsContainer.empty();
+
+        // Reset the letter-filter if click on the current selected letter
+        letter = ''
+        if ( !$this.hasClass('current') ){
+            letter = $this.data('key')
+        }
+        reloadView();
+    }
+    var setMoreButtonVisible = function(setVisible){
+        if ( setVisible ){
+            $loadMoreButton.show();
+        }else{
+            $loadMoreButton.hide();
+        }
+    }
+    var updateSearch = function(text){
+        index = 0;
+        $contactsContainer.empty();
```

Comment:
This will "glitch" when two updates are triggered and the second update is triggered before the first one receives its contents, which will result in:

- updateSearch()
- empty()
- reloadView()
- do AJAX # 1 (this means async :wink)
- updateSearch()
- empty()
- reloadView()
- do AJAX # 2 (this means async :wink)
- receive AJAX # 1 -> append (!) to content
- receive AJAX # 2 -> append (!) to content

then you have duplicate entries:
--
<img width="412" alt="bildschirmfoto 2015-08-20 um 15 56 33" src="https://cloud.githubusercontent.com/assets/7469/9384971/0dcc739e-4754-11e5-8f9d-334dbdd58bad.png">
--

Can easily be reproduced by:
- Copying the name of the contact
- Pasting it to the search field multiple times in a row very fast (e.g. `⌘-a ⌘-v`, repeat)